### DISCUSSION TITLE
Confusing method names in LocalParameters?

### DISCUSSION THREAD

**Discussion by thomascorthals** [solariumphp/solarium#917] [Ideas]
#884 got me wondering about this:

https://github.com/solariumphp/solarium/blob/a5bb12e15ee5eef7cc6835cd4bc6fb50357e76e1/src/Core/Query/LocalParameters/LocalParameters.php#L98-L132

`setExclude()` actually adds one exclude, the same way `addExcludes()` can add multiple excludes, instead of overwriting previously set excludes the way `setExcludes()` does. This isn't intuitive and not mentioned in the docs.

**Comment by thomascorthals** [comment]
This is a better example of how it's confusing because it directly shows the inconsistency in the usage of add vs. set.

https://github.com/solariumphp/solarium/blob/ada911c3eee6023d4ed7e16b80c8e0a8c43c0664/src/QueryType/Select/Query/FilterQuery.php#L66-L92

**Comment by mkalkbrenner** [comment]
@thomascorthals 
In general "add" should add something to the existing list. "set" should empty the existing list and set the new value(s).
So you're right and it looks like a bug at the moment.
